### PR TITLE
Run anywhere

### LINKED ISSUE #1: Support running this right after creating a tag in a workflow
I'm opening this issue to keep track of this feature as discussed in https://github.com/fregante/github-url-detection/pull/30 and https://github.com/sindresorhus/refined-github/pull/3421

Currently this action only supports the `on.push.tag` event

### PR DESCRIPTION
**Pull request by notlmn**
Closes #1 

- Run on events other than pushing tags
- Allow user to pass in a tag input
- Take token from input instead of `ENV`

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -24,8 +24,7 @@ jobs:
         footer: 'Custom footer'
         include-hash: true
         include-range: true
-      env:
-        RELEASE_TOKEN: ${{ secrets.RELEASE_TOKEN }}
+        token: ${{ secrets.RELEASE_TOKEN }}
```
Comment (fregante) [inline readme.md]: I think this should use the default token

File: distribution/index.js [hunk 2]
```diff
@@ -170,12 +170,6 @@ const execFile = util.promisify(__webpack_require__(129).execFile);
```
Comment (fregante) [inline distribution/index.js:170]: `distribution` should have the [`linguist-generated`](https://github.com/github/linguist#generated-code) git attribute so its diffs are collapsed

File: index.js [hunk 3]
```diff
@@ -22,16 +16,25 @@ async function run() {
 		const {stdout: t} = await execFile('git', ['tag', '-l', '--sort=-creatordate']);
 		const tags = t.split('\n').filter(Boolean).map(tag => tag.trim());
 
+		if (tags.length === 0) {
+			core.setOutput('There is nothing to be done here. Exiting!');
```
Comment (fregante) [inline index.js]:
```suggestion
			core.info('There is nothing to be done here. Exiting!');
			return;
```

The output should not be a message, but just data, if any: https://github.com/actions/toolkit/tree/master/packages/core#inputsoutputs

File: index.js [hunk 4]
```diff
@@ -22,16 +16,25 @@ async function run() {
...
+		let tag = core.getInput('tag') || tags[0];
+
 		// Warn users of tags out of order / for pushing older tags
-		if (pushedTag !== tags[0]) {
-			core.warning('Looks like you may be pushing outdated tags. Make sure you are pushing the right tags!');
+		if (process.env.GITHUB_REF.startsWith('refs/tags/') === true) {
+			tag = process.env.GITHUB_REF.replace('refs/tags/', '');
+			if (tag !== tags[0]) {
+				core.warning('Looks like you may be pushing outdated tags. Make sure you are pushing the right tags!');
+			}
```
Comment (fregante) [inline index.js]:
```suggestion
```

I don't think this is a problem

File: index.js [hunk 5]
```diff
@@ -22,16 +16,25 @@ async function run() {
...
+		if (tags.length === 0) {
+			core.setOutput('There is nothing to be done here. Exiting!');
+		}
+
+		let tag = core.getInput('tag') || tags[0];
+
 		// Warn users of tags out of order / for pushing older tags
-		if (pushedTag !== tags[0]) {
-			core.warning('Looks like you may be pushing outdated tags. Make sure you are pushing the right tags!');
+		if (process.env.GITHUB_REF.startsWith('refs/tags/') === true) {
```
Comment (fregante) [inline index.js]:
```suggestion
		if (process.env.GITHUB_REF.startsWith('refs/tags/')) {
```

File: index.js [hunk 6]
```diff
@@ -22,16 +16,25 @@ async function run() {
 		const {stdout: t} = await execFile('git', ['tag', '-l', '--sort=-creatordate']);
 		const tags = t.split('\n').filter(Boolean).map(tag => tag.trim());
 
+		if (tags.length === 0) {
```
Comment (fregante) [inline index.js:22]: Currently it doesn't seem to support creating a release for the first tag (which is usually just "Initial release"), so this condition should reflect that, at least for the time being.

```suggestion
		if (tags.length < 2) {
```

**Review by fregante — COMMENTED** [review]
I don't have time to test this, but if you add it to RGH even as a `user/action#commit` we could test it tomorrow.

File: index.js (hunk 6 above)
Comment (fregante) [reply index.js:22]: This is unchanged. Did you add support for "initial release" in the commit?

Comment (notlmn) [reply index.js:22]: Support for "initial release" has been there the whole time.

If there is only one tag, then we compare that with the root commit (initial commit) we can find.

image

**Comment by fregante** [comment]
Looks good, but can test this PR on https://github.com/sindresorhus/refined-github/pull/3421, https://github.com/fregante/github-url-detection/pull/30 or https://github.com/notlmn/browser-extension-template/issues/37?

So we're sure it works the way we want it to work.

File: index.js (hunk 6 above)
Comment (fregante) [reply index.js:22]: I'd prefer this to be just `Initial release` but that works too, for the time being

**Comment by notlmn** [comment]
Will try this on https://github.com/notlmn/browser-extension-template/issues/37, as that looks like the best place to test this stuff without polluting other repositories.

**Review by fregante — APPROVED** [review]

File: index.js [hunk 7]
```diff
@@ -66,7 +72,7 @@ async function run() {
 			releaseBody.push(`\n[\`${range}\`](https://github.com/${owner}/${repo}/compare/${range})`);
 		}
 
-		const octokit = getOctokit(process.env.RELEASE_TOKEN);
+		const octokit = getOctokit(core.getInput('token'));
```
Comment (fregante) [inline index.js:75]: You can use this package to automatically pickup the token: https://github.com/octokit/action.js

The readme doesn’t show with.token support but I just added it in a dependency: https://github.com/octokit/auth-action.js/pull/106

Comment (notlmn) [reply index.js:75]: That sounds cool, do you want me to add this to #22?

Comment (fregante) [reply index.js:75]: Doesn’t sound related 😃

**Pull request merged**